Feature/ccp 248 (Migrate to SQL)

kkrish39 opened the pull request:
Unable to open the previous PR as we deleted CCP-254 branch (I did raise the PR against CCP-254 before). Creating a new PR now to merge to develop.

Previous PR link - https://github.com/diging/conceptpower/pull/208

kkrish39 committed:
[CCP-253] No Logics. Only HTML section Indendation.

kkrish39 committed:
[CCP-253] Added Methods to return All review Requests

kkrish39 committed:
[CCP-253] Added a section to feed in resolve comment and respective
Backend Changes

kkrish39 committed:
[CCP-253] Added ResolveRequest feature and re addition a new request for
the resolved request

kkrish39 committed:
[CCP-253] Added reopening the resolved request logic

kkrish39 committed:
[CCP-251, CCP-252, CCP-253] Minor Re-factoring and Adding JavaDoc

kkrish39 committed:
[CCP-251, CCP-252, CCP-253] Test case for Update and Reopen Review
Request

kkrish39 committed:
[CCP-251, CCP-252, CCP-253] Fixing trailing spaces on review request and
comment textbox

kkrish39 committed:
Merge branch 'develop' and added security tokens for reviewconcept's url

kkrish39 committed:
[CCP-254] Adding ThymeLeaf Configuration

kkrish39 committed:
[CCP-253] Fix - Save history of review Requests and Comments

kkrish39 committed:
[CCP-253] Enhancement to view the review History. CSS needs to be fixed

kkrish39 committed:
[CCP-253] Render history of ReviewRequests in UI

kkrish39 committed:
[CCP-253] Minor Refactoring on CSS and Alignment

kkrish39 committed:
[CCP-253] Fixed Test Cases

kkrish39 committed:
[CCP-253] Revamped the logic of storing and displaying requests and
comments

kkrish39 committed:
[CCP-253] Added Testcases for RequestManager

kkrish39 committed:
[CCP-253] Minor Refactoring and ReviewRequestClient testcases

adhvikmadhav committed:
[CCP - 254] Adding NavBar for HomePage Layout

kkrish39 committed:
[CCP-253] Validating the data sent from the client and minor refactoring

adhvikmadhav committed:
[CCP-254] Added search and footerSection of the home page

adhvikmadhav committed:
[CCP 254] Pagination and subtle fixes

kkrish39 committed:
[CCP-253] Added event handlers to search and dropdown

kkrish39 committed:
[CCP-254] Segregated the page components

adhvikmadhav committed:
[CCP-254] Added four new pages

kkrish39 committed:
[CCP-254] Fixed Indendation

kkrish39 committed:
[CCP-254] Added Modal to view concept details

adhvikmadhav committed:
[CCP-254] Built 4 new pages

kkrish39 committed:
[CCP-253] Addressing pull request comments

adhvikmadhav committed:
[CCP-254] Final static webpages

adhvikmadhav committed:
[CCP-254] exception page

kkrish39 committed:
Addressing Pull request review comments

kkrish39 committed:
[CCP-253] Revoking the settings file changes

kkrish39 committed:
[CCP-254] Coupled Concept search button with endpoint

kkrish39 committed:
[CCP-253] Reused getAllReview code and removed List of comments

kkrish39 committed:
[CCP-253] Address PR review Comments and corresponding test case fixes

kkrish39 committed:
[CCP-254] Integrated Pagination, Modal and Segregated Modals

kkrish39 committed:
[CCP-254] Changed the concept Details modals from being static to dynamic addition

kkrish39 committed:
[CCP-254] Included Routing from homepage and excluded auth route for testing

kkrish39 committed:
[CCP-254] Added Spinner

kkrish39 committed:
[CCP-254] Added sorting feature for table data and minor tweaks in pagination

kkrish39 committed:
[CCP-254] Typo in event handler

kkrish39 committed:
[CCP-254] Added CSS to show mouse pointer

kkrish39 committed:
[CCP-254] Restored the removed auth routes and integrated login/logout feature

kkrish39 committed:
[CCP-254] Added Edit/Delete Concept api and integrated conceptTypeAddForm

adhvikmadhav committed:
[CCP 254] New pages and templates

kkrish39 committed:
[CCP-253] Serialized OffsetDateTime class while saving

kkrish39 committed:
[CCP-254] Removed all redundant Imports and added main html

adhvikmadhav committed:
[CCP 254] Modals, small UI changes, New web page

kkrish39 committed:
[CCP-254] Coulped form error fields to the UI and minor refactoring

adhvikmadhav committed:
[CCP-254] Index mgmnt page, edit user details, password reset

kkrish39 committed:
[CCP-254] Merge concept feature(Requires Refactoring)

kkrish39 committed:
[CCP-253] Addressing Pull Request Comments

adhvikmadhav committed:
[CCP-254] Adding and deleting users, changes to home page

kkrish39 committed:
[CCP-254] Merge concept related changes

kkrish39 committed:
[CCP-253] Fixing test cases

kkrish39 committed:
[CCP-253] Added custom ObjectSet classes

kkrish39 committed:
[CCP-254] Index Management Page

kkrish39 committed:
[CCP-254] Integrated ConceptList and AddConceptList Page

kkrish39 committed:
[CCP-254] Details of merged concepts API and Add concept view

adhvikmadhav committed:
[CCP-254] Editing and deleting concept lists types, Fixed Add concept
page

kkrish39 committed:
[CCP-254] Renamed files

kkrish39 committed:
[CCP-254] Added synonym and term search  modules in AddConcept page

adhvikmadhav committed:
[CCP-254] Adding concepts, change passwords for non-admin users,
concepts review

adhvikmadhav committed:
[CCP 254] Tiny UI Changes

kkrish39 committed:
[CCP-253] Handling Null Exception for concept createdAt

adhvikmadhav committed:
[CCP-254] Editing concepts page(same IDs), deleting concepts from
homepage(incomplete)

kkrish39 committed:
[CCP-253] handling null values for reviewRequest

kkrish39 committed:
[CCP-254] Null check for comments list

adhvikmadhav committed:
[CCP-254] DEleting concepts, fixed sign in issues

kkrish39 committed:
[CCP-254] Merge Concept feature

adhvikmadhav committed:
[CCP-254] minor usability changes

adhvikmadhav committed:
[CCP-254] redirecting to edit concept page

kkrish39 committed:
[CCP-254] merge concept refactoring

adhvikmadhav committed:
[CCP-254] Fixed drop-downs, populated edit concepts page, synonyms in
progress

kkrish39 committed:
[CCP-254] Refactoring Merge concept ang Add Concept Page

kkrish39 committed:
[CCP-254] Refactoring EditConcept page/Replacing tags with th:field

adhvikmadhav committed:
[CCP-254] Fixed super type issue in Concept types, fixed synonyms issue
in edit concept, small UI changes

jdamerow committed:
[CCP-254] fixed search issue for wordnet concepts

adhvikmadhav committed:
[CCP-254]

adhvikmadhav committed:
[CCP-254] Lucehen Utility changes

kkrish39 committed:
[CCP-254] Modifications in add concept wrapper page

kkrish39 committed:
[CCP-254] WordNetWrapper Modifications

adhvikmadhav committed:
[CCP-254]

kkrish39 committed:
[CCP-254] Number of records per page feature

kkrish39 committed:
[CCP-254] Code Refactoring

kkrish39 committed:
[CCP-254] Added CSS for form alerts

kkrish39 committed:
[CCP-254] CSS Refactoring

kkrish39 committed:
[CCP-254] Removing Concept edit redirection check from controller

kkrish39 committed:
[CCP-254] Added redirection check in homepage

kkrish39 committed:
[CCP-254] Add/Edit Concept Wrapper Integeration

kkrish39 committed:
[CCP-254] Fix: Delete concepts - Reverted luceneutility changes

kkrish39 committed:
[CCP-254] CSS Refactoring in AddConceptWrapper page and ConceptDetails modal

kkrish39 committed:
Added User Profile and Change Password Page

kkrish39 committed:
[CCP-254] CSS Refactoring

kkrish39 committed:
[CCP-254] Changed the file names to lowercase

kkrish39 committed:
[CCP-254] reset password/email/reseterror pages refactoring

kkrish39 committed:
[CCP-254] Renaming filenames in controller and removing console statements

kkrish39 committed:
[CCP-254] Removing popup for edit and delete concepts/users/conceptlists and Minor code refatoring

kkrish39 committed:
[CCP-254] Code Refactoring

kkrish39 committed:
[CCP-254] Added concept Details to delete popup in homepage

Commit:
[CCP-254] Added Concept Details page and refactored add concept page

Commit:
[CCP-254] Removed Tiles related configurations

Commit:
[CCP-254] Deleted all .jsp and tile config files

Commit:
[CCP-254] Code Refactoring (Indendation)

Commit:
[CCP-254] restored settings file

Commit:
[CCP-254] Added pagination to Concepttypes page

Commit:
[CCP-254] Addressing pull request comments

Commit:
[CCP-254] Addressing PR reviews - Spacing/table

Commit:
[CCP-254] Changed layout color

Commit:
[CCP-254] Addressing PR Comments(Button Size/Index 404)

Commit:
[CCP-254] Addressing pull request (logo/pagewidth)

Commit:
[CCP-254] Restoring changes in conceptTypeEditController

Commit:
[CCP-254] Removing concept details controller

Commit:
Addressing Pull Request Comments(details/siging/Pagination/button/synonym/descrpiton)

Commit:
[CCP-254] Addressing PR comments(Indexing/DeleteConcept/AdminIconColor)

Commit:
[CCP-254] Addressing PR comments(Review Request/Delete Icon/Form submit)

Commit:
[CCP-254] Addressing PR Comments except for the addConceptWrapper Page

Commit:
[CCP-254] Complete refactoring of addconceptwrapper page

Commit:
[CCP-254] Addressing PR Comments(NavabarRearrange/DeleteColor/ErrorMsg/InputShape/whiteSpaces)

Commit:
[CCP-254] Addressing pull request comments(Add/Remove Synonym)

Commit:
[CCP-254] Addressing PR comments(Merge Concept)

Commit:
[CCP-254] Minor CSS Refactoring

kkrish39 committed:
[CCP-248-2] Added base folder/files and dependancies

kkrish39 committed:
[CCP-254] Addressing Pull request comments(Version/NumRecords dropdown

kkrish39 committed:
[CCP-254] Addressing pull request comments(Modal/Signin/dataTables)

kkrish39 committed:
[CCP-254] Addressing PR comments(handling cancel in editConcept/pagination)

kkrish39 committed:
[CCP-248-2] Pushed skleton files

kkrish39 committed:
Removing DS_Store file

kkrish39 committed:
Converting the build data/PR data to raw string

kkrish39 committed:
Fixed noConcept display bug/Show concept

kkrish39 committed:
Fixed all the PR comments

kkrish39 committed:
[CCP-254] - Chaning pom.xml

kkrish39 committed:
[CCP-254] Changes to footer

kkrish39 committed:
[CPP-248-2] Added a method

kkrish39 committed:
[CCP-254] - Fixed all the Pull request comments

kkrish39 committed:
Merge branch 'feature/CCP-254' of https://github.com/diging/conceptpower into feature/CCP-248-2

kkrish39 committed:
[CCP-248-2]Configured with sql instance and added a basic UI

kkrish39 committed:
[CCP-248-2]Added Model and and implementations for ReviewRequest class

kkrish39 committed:
Merge branch 'feature/CCP-253' of https://github.com/diging/conceptpower into feature/CCP-248-2

kkrish39 committed:
[CCP-248-2] Configuring to migrate concept types entity to sql

kkrish39 committed:
[CCP-254] Addressing PR comments

kkrish39 committed:
[CCP-254] Fix: Details are not getting displayed in the modal

kkrish39 committed:
Merge branch 'feature/CCP-254' of https://github.com/diging/conceptpower into feature/CCP-248-2

kkrish39 committed:
[CCP-248-2] Updated HikariCP version

kkrish39 committed:
[CCP-248-2] Added Repository/Services/Models to migrate reviewrequest table

kkrish39 committed:
[CCP-248-2]Removed Object DB Dependencies

kkrish39 committed:
[CCP-248-2] Refactored ReviewRequest table migration

kkrish39 committed:
[CCP-248-2] Added Comment Table migration logic

kkrish39 committed:
[CCP-248-2] Added ConceptType Table migration related files

kkrish39 committed:
[CCP-248-2] Minor Refactoring

kkrish39 committed:
[CCP-248-2] Added temporary script file to keep track fo the sql queries

kkrish39 committed:
[CCP-248-2] Added ConceptList Table migration related files

kkrish39 committed:
[CCP-254] Fix: fixed square bracket getting displayed in equalsValues

kkrish39 committed:
[CCP-254] Fix: fixed synonym modal issues in add concept page

kkrish39 committed:
[CCP-254] Fix: Concepts to be merged is not getting displayed when there is an error while merging

kkrish39 committed:
[CCP-248-2] Added ConceptList migration

kkrish39 committed:
[CCP-248-2] Added method to get all reviewRequests in the DB

kkrish39 committed:
[CCP-248-2] Added methods to get all concepts

kkrish39 committed:
[CCP-248-2] Removing all Interfaces for getters and setters

kkrish39 committed:
[CCP-248-2] Refactored all entities

kkrish39 committed:
[CCP-248-2] Refactored all Repository classes and interfaces

kkrish39 committed:
[CCP-248-2] Refactored all service classes and interfaces

kkrish39 committed:
[CCP-248-2] Refactored all Migration related files

kkrish39 committed:
Merge branch 'feature/CCP-254' of https://github.com/diging/conceptpower into feature/CCP-248-2

kkrish39 committed:
[CCP-248-2] Refactored UI components

kkrish39 committed:
Added springframework pagingrepo interface

kkrish39 committed:
Updated spring version and spring-data-jpa version

kkrish39 committed:
Addressing PR comments and Minor Refactoring

kkrish39 committed:
[CCP-248-2] Minor Refactoring of pom file

kkrish39 committed:
Addressing pull request comments

kkrish39 committed:
Added Name tags in all files

kkrish39 committed:
Merge branch 'develop' of https://github.com/diging/conceptpower into feature/CCP-248-2

diging-jenkins commented:
Can one of the admins verify this patch?

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented on Conceptpower+Spring/pom.xml:
```diff
@@ -1,800 +1,783 @@
+      <db.database.url></db.database.url>
+        <db.user></db.user>
+        <db.password></db.password>
+      <hibernate.dialect>org.hibernate.dialect.MySQL5InnoDBDialect</hibernate.dialect>
```
indentation

jdamerow commented on Conceptpower+Spring/pom.xml:
```diff
@@ -1,800 +1,783 @@
+      <skip.unit.tests>false</skip.unit.tests>
+      <concept.list.url>https://www.dropbox.com/s/r82nymnt2247zz4/conceptLists.db?raw=1</concept.list.url>
+      <concept.types.url>https://www.dropbox.com/s/nwn15s84fzz37s5/conceptTypes.db?raw=1</concept.types.url>
+      <users.url>https://www.dropbox.com/s/p6h3974hmlouq0i/users.db?raw=1</users.url>
```
we gotta fix the integration tests; I think for now I would add test database configs and fill the database before running the test cases and empty it afterwards. Not ideal but will probably work.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/model/impl/ReviewRequest.java:
```diff
@@ -0,0 +1,118 @@
+    private String request;
+    
+    @Column(name="status")
+    private ReviewStatus status;
```
you probably want to use the `Enumerated` annotation here and set it to use the string representation for the enum. I think the default is the index, which easily introduces bugs when whoever makes changes to the enum forgets and changes the order. With string values at least it's easy to spot when looking at the database.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db4o/IRequestsDBManager.java:
```diff
@@ -1,13 +1,23 @@
+
+/**
+ * 
+ * @author Keerthivasan
```
full name please for javadoc ;)

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/ConceptManager.java:
```diff
@@ -625,4 +625,9 @@ public ConceptEntry getOriginalConceptEntry(String id) {
 
         return null;
     }
+
```
this class still uses the db4o database, rigth? It needs to use the new sql classes so it serves up the migrated data. This is true for all manager classes.

kkrish39 committed:
[CCP-248] Added full-names to author tag

kkrish39 committed:
[CCP-248] Added enumeration tags

kkrish39 committed:
Added SQL version of concept list transactions

kkrish39 commented:
Added only Concept List. Reopening PR just to make sure I'm proceeding in right track.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/ConceptListService.java:
```diff
@@ -0,0 +1,78 @@
+
+    @Override
+    public void deleteConceptList(String name) {
+        // TODO: Must check whether should we proceed with delete by conceptList name
```
No, since we should now have ids for all lists and we're migrating anyways, we should use IDs for everything.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/ConceptListService.java:
```diff
@@ -0,0 +1,78 @@
+            // if there doesn't exist an object with this id return id
+            Optional<ConceptList> results = conceptListRepository.findById(id);
+            if (!results.isPresent())
+                return id;
```
please follow our style guide

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/ConceptListService.java:
```diff
@@ -0,0 +1,78 @@
+        return conceptListRepository.findByConceptListName(name) != null;
+    }
+    
+    protected String generateId(String prefix) {
```
actually, let's switch to database generated ids for lists at least. If you check one of the other projects (citesphere I believe has it for example, or virtual spaces), there is id generated that uses a prefix and then uses a database counter to generate ids.

kkrish39 committed:
[CCP-248] Added changes to use ConceptType SQL Manager

kkrish39 committed:
[CCP-248] Moved all repost to repository package

kkrish39 committed:
[CCP-248] Moved all models out of core pacakege

kkrish39 committed:
[CCP-248] Changed config to process entities

kkrish39 committed:
[CCP-248] Moved manager files out of core package

kkrish39 committed:
[CCP-248] Added Manager class for reviewRequest table

kkrish39 committed:
[CCP-248] wiring new concept entry model

kkrish39 committed:
[CCP-248] Modified all the files with new SQL Entities

kkrish39 committed:
Revert "[CCP-248] Added changes to use ConceptType SQL Manager"

This reverts commit d37d5c02b90dfaec3a27cf938560fad54a90c1cc.

kkrish39 committed:
[CCP-248] Revoked a couple of previous migration changes

kkrish39 committed:
Renamed IConceptTypeManger to IConceptTypeManager

kkrish39 committed:
[CCP-248] Replaced the database client methods with SQL Repository

kkrish39 committed:
[CCP-248-2] Fixed unit test cases

kkrish39 committed:
[CCP-248-2] Fixed ordinal issue to save enum values

kkrish39 committed:
[CCP-248-2] Added Specification configuration to search custom fields

kkrish39 committed:
[CCP-248-2] Fixed unmodifiable list sort

kkrish39 committed:
[CCP-248-2] Fixed add concept and wrapper issues

kkrish39 committed:
[CCP-248-2] Removing print statement and added a null check

kkrish39 commented:
Reopening the PR, with an existing Integration test failure to check if it's occurring while deployment as well. (Related to csrf)

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/TypeDatabaseClient.java:
```diff
@@ -1,153 +1,83 @@
+    public List<ConceptType> getAllTypes(int page, int pageSize, final String sortBy, int sortDirection){
+        page = page < 0 ? 0 : page;
+
+        pageSize = pageSize == -1 ? defaultPageSize : pageSize;
```
you probably want to check for <= 0 here to avoid making a call with an invalid page size (same thing above)

jdamerow replied:
?

kkrish39 replied:
Umm, this is tricky. The change is already in the branch. If you open the file separately, you can find the updated code. Not sure why it's not reflected here.

jdamerow replied:
This version has -1 not 0:
https://github.com/diging/conceptpower/blob/feature/CCP-248-2/Conceptpower%2BSpring/src/main/java/edu/asu/conceptpower/app/db/TypeDatabaseClient.java#L73
Maybe you didn't push it?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/manager/RequestsManager.java:
```diff
@@ -0,0 +1,116 @@
+import edu.asu.conceptpower.app.model.ReviewStatus;
+
+@Service
+public  class RequestsManager implements IRequestsManager{
```
Is there a reason only the implementation is in this package and not the interface? (implementation should be in an impl package in the package with the interface).

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/model/ConceptEntry.java:
```diff
@@ -0,0 +1,292 @@
+         return this.changeEvents;
+    }
+   
+    public void setChangeEvents(ChangeEvent changeEvent) {
```
`setChangeEvents` should be a proper setter method for `changeEvents`. This method should be `addChangeEvent` since it just adds an event.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/ConceptEntrySpecification.java:
```diff
@@ -0,0 +1,33 @@
+
+public class ConceptEntrySpecification {
+
+    public static Specification<ConceptEntry> customFieldSearch(String fieldName, String fieldQuery) {
```
why is this a static method?

jdamerow replied:
Since this is only used in one place, why dont' you build the query right where it is needed?

jdamerow commented on Conceptpower+Spring/src/main/resources/config.properties:
```diff
@@ -3,4 +3,18 @@ lucenePath=${lucene.path}
+db.url=jdbc:mysql://localhost:3306/testdb?useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC
+
+db.username=root
+db.password=password
```
db properties need to be configurable through pom

jdamerow commented on Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/db/customobjects/PredicateObjectSet.java:
```diff
@@ -0,0 +1,39 @@
+        //
+    }
+    
+}
```
what are these classes for (that import db4o)? the tests shouldn't use db40 anymore.

jdamerow reviewed (changes requested):
I don't see any integration test adjustment (setting up/removing db tables, etc)? Since Taylor mentioned it earlier regarding Vogon, the easiest way is probably to have a SQLlight file we use instead of the db4o files.

kkrish39 committed:
[CCP-248-2] Moved Manager classes into a separate package

kkrish39 committed:
[CCP-248-2] Renaming changeEvent setter

kkrish39 committed:
[CCP-248] Added user controller table in My SQL format

kkrish39 committed:
[CCP-248] Moved database properties to POM

kkrish39 committed:
[CCP-248] Added User Table Migration

kkrish39 committed:
[CCP-248-2] Added SQLite integration

kkrish39 committed:
[CCP-248-2] Reorganized pom.xml file

kkrish39 committed:
[CCP-248-2] Minor Refactoring

kkrish39 committed:
[CCP-248-2] Fixed Integration test failures

kkrish39 committed:
[CCP-248-2] Added SQLLite In-mem db

kkrish39 committed:
[CCP-248-2] Moved Indexing event table from db40 to MySQL

kkrish39 committed:
[CCP-248-2] Updated SQLLite db

kkrish39 committed:
[CCP-248-2] Updated SQLLite db

kkrish39 committed:
[CCP-248-2] Updated SQLLite db

kkrish39 committed:
[CCP-248-2] Minor ChangeEvent table changes

kkrish39 committed:
[CCP-248-2] Fix: JoinColumn Issue between changeevent and ConceptEntry

kkrish39 committed:
[CCP-248-2] Fixed integration tests

kkrish39 committed:
[CCP-248-2] Minor Refactoring and Resolving Conflicts

kkrish39 committed:
[CCP-248-2] Minor Removed Unused Imports

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/config/PersistentContext.java:
```diff
@@ -0,0 +1,85 @@
+        JpaTransactionManager transactionManager = new JpaTransactionManager();
+        transactionManager.setEntityManagerFactory(entityManagerFactory);
+        return transactionManager;
+    }
```
is there a reason the above two methods are not public?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/DatabaseClient.java:
```diff
@@ -389,14 +231,7 @@ public void update(ConceptEntry entry, String databasename) {
-            dictionaryClient.delete(res);
-            dictionaryClient.commit();
-        }
+        conceptListRepository.deleteByConceptListName(name);
```
didn't it change that concept lists were identified by their name? Don't they have ids now? if so, ids should be used to delete a list.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/DatabaseClient.java:
```diff
@@ -412,9 +247,29 @@ public void update(ConceptList list, String listname, String databasename) {
+    
+    @Override
+    public void storeConceptList(ConceptList element, String databasename) {
+        if (databasename.equals(DBNames.DICTIONARY_DB)) {
```
is this still necessary?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneDAO.java:
```diff
@@ -61,26 +51,19 @@ public IndexingEvent getTotalNumberOfWordsIndexed() {
-                latestTimeStamp = t;
-            }
+            
+            latestTimeStamp = t;
```
merge with line 60

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/manager/impl/ConceptListManager.java:
```diff
@@ -91,14 +91,8 @@ protected String generateId(String prefix) {
-			List<Object> results = client.queryByExample(example);
-			if (results == null || results.size() == 0)
+			boolean result = client.checkIfConceptListExists(id);
+			if (!result)
```
merge with line 94

jdamerow replied:
merge with line 93 and follow style guide

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/manager/impl/ConceptManager.java:
```diff
@@ -63,6 +64,8 @@
     @Autowired
     private IAlternativeIdService alternativeIdService;
+    
+    SimpleDateFormat ft = new SimpleDateFormat ("yyyy:MM:dd hh:mm:ss");
```
let's make that a configuration property (the date format)

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/manager/impl/ConceptManager.java:
```diff
@@ -495,10 +498,10 @@ public ConceptEntry addConceptListEntry(ConceptEntry entry, String userName) thr
         changeEvent.setUserName(userName);
         changeEvent.setType(ChangeEventTypes.MODIFICATION);
-        entry.addNewChangeEvent(changeEvent);
+       entry.addChangeEvent(changeEvent);
```
there are some indentation issues in the above lines

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/ConceptEntrySpecification.java:
```diff
@@ -0,0 +1,33 @@
+        private String getLikePattern(final String searchTerm) {
+            StringBuilder pattern = new StringBuilder();
+            pattern.append(searchTerm.toLowerCase());
+            pattern.append("%");
```
this will search startsWith not contains, does it? But don't you need a contained query?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/validation/ConceptListAddValidator.java:
```diff
@@ -35,7 +35,7 @@ public void validate(Object target, Errors errors) {
             if (!conceptListAddForm.getListName().equalsIgnoreCase(conceptListAddForm.getOldListName())) {
 
-                if (conceptListManager.checkExistingConceptList(conceptListAddForm.getListName())) {
+                if (conceptListService.checkExistingConceptList(conceptListAddForm.getListName())) {
```
this is not a requirement anymore if there are IDs now

kkrish39 committed:
[CCP-248] Changed conceptList to find through id

kkrish39 committed:
[CCP-248] Moved manager classes/Minor Refactoring

kkrish39 committed:
[CCP-248] Removed DBNames class

diging-jenkins commented:
Try creating a new PR, sometimes there are issues with the PRs if branches change.

---
Julia Damerow, PhD
Scientific Software Engineer
Global Biosocial Complexity Initiative
Digital Innovation Group
Arizona State University
email: jdamerow@asu.edu <>
web: http://diging.asu.edu, https://complexity.asu.edu/















> On Jan 6, 2021, at 1:56 PM, Keerthivasan Krishnamurthy <notifications@github.com> wrote:
> 
> 
> @kkrish39 commented on this pull request.
> 
> In Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/TypeDatabaseClient.java <https://github.com/diging/conceptpower/pull/211#discussion_r552903096>:
> 
> > -        };
> -
> -        typeQuery.sortBy(conceptTypeComparator);
> -        List<ConceptType> conceptTypes = typeQuery.execute();
> -
> -        int startIndex = (page - 1) * pageSize;
> -        int endIndex = startIndex + pageSize;
> -        if (endIndex > conceptTypes.size()) {
> -            endIndex = conceptTypes.size();
> -        }
> -
> -        return new ArrayList<ConceptType>(conceptTypes.subList(startIndex, endIndex));
> +    public List<ConceptType> getAllTypes(int page, int pageSize, final String sortBy, int sortDirection){
> +        page = page < 0 ? 0 : page;
> +
> +        pageSize = pageSize == -1 ? defaultPageSize : pageSize;
> Umm, this is tricky. The change is already in the branch. If you open the file separately, you can find the updated code. Not sure why it's not reflected here.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/diging/conceptpower/pull/211#discussion_r552903096>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADEDRQIVCTXBQ3UMLHLMXDDSYSW67ANCNFSM4RN3VEQA>.
>

kkrish39 committed:
[CCP-248] Moved Date format to configuration

kkrish39 committed:
[CCP-248] Addressing at PR comments

kkrish39 committed:
[CCP-248] Fixed all test cases/Minor Refactoring

kkrish39 commented:
I've made a simple work around fix to handle the Unit test case failure instead of configuring it with the environment properties.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/DatabaseClient.java:
```diff
@@ -407,14 +221,50 @@ public void deleteConceptList(String name) {
-            dictionaryClient.store(list);
-            dictionaryClient.commit();
-        }
+    public void update(ConceptList list, String listname) {
```
this should now get the list by id, not by name.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/model/Comment.java:
```diff
@@ -0,0 +1,82 @@
+    private OffsetDateTime createdAt;
+    
+    @ManyToOne
+    // @JoinColumn(name = "review_request_id", referencedColumnName="concept_id")
```
remove if not needed

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/ITokenRepository.java:
```diff
@@ -0,0 +1,22 @@
+    
+    Token findByToken(String token); 
+    
+    @Transactional
```
I don't think that'll do much here. Transactional annotation should be on the manager classes.

jdamerow replied:
?

kkrish39 replied:
I included the @Transactional annotation for the calling method as well. Shouldn't that be enough to handle the transactional failure? 

In this file - https://github.com/diging/conceptpower/blob/feature/CCP-248-2/Conceptpower%2BSpring/src/main/java/edu/asu/conceptpower/app/users/UserDatabaseClient.java

jdamerow replied:
You only need one transactional annotation (or multiple if you do several methods in a class, but as a class annotation makes in most cases sense). If you have it on the manager class, you don't need it here.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/IUserRepository.java:
```diff
@@ -0,0 +1,30 @@
+    
+    List<User> findAll();
+    
+    @Transactional
```
see above

jdamerow replied:
?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/users/UserDatabaseClient.java:
```diff
@@ -2,41 +2,32 @@
+    private IUserRepository client;
+    
+    @Autowired
+    private ITokenRepository tokenClient;
```
let's call the repo variables repos, e.g. `tokenRepo` or `tokenRepository`

kkrish39 committed:
[CCP-248] Addresing all PR comments

kkrish39 commented:
I'm not sure if there was an issue while pushing. I can see -1 in the old commit as well.

This should be the commit - https://github.com/diging/conceptpower/commit/47e0a47be2938fc3f732c7b89253915b4bb316f3

kkrish39 committed:
[CCP-248] Minor Refactoring

kkrish39 committed:
[CCP-248] Typo in indexing event constructor

kkrish39 committed:
[CCP-248] Fix:bug in pageSize initialization

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/DatabaseClient.java:
```diff
@@ -212,209 +132,177 @@ public boolean match(ConceptEntry con) {
-        }
-
-        List<ConceptEntry> dictResults = dictQuery.execute();
+        pageSize = pageSize == -1 ? defaultPageSize : pageSize;
```
shouldn't that be `< 0` to make sure it's not an invalid page size?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/TypeDatabaseClient.java:
```diff
@@ -1,153 +1,83 @@
+    public List<ConceptType> getAllTypes(int page, int pageSize, final String sortBy, int sortDirection){
+        page = page <= 0 ? 0 : page;
+
+        pageSize = pageSize == 0 ? defaultPageSize : pageSize;
```
`< 0`?

jdamerow replied:
?

jdamerow replied:
?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/manager/impl/AlternativeIdService.java:
```diff
@@ -36,23 +36,23 @@ public void addAlternativeIds(String queriedId, ConceptEntry entry) {
             for (String mergedId : mergedIds) {
-                entry.getAlternativeIds().add(mergedId.trim());
+               entry.setAlternativeIds(mergedId.trim());
             }
```
why are these substitutions made?

kkrish39 replied:
The main reason was to avoid having a new table for alternative Id's in the database. I see the bug now.

jdamerow replied:
?

jdamerow replied:
this is still here. why is this setting the id erverytime?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/migration/impl/MigrateToSql.java:
```diff
@@ -0,0 +1,150 @@
+        for(ConceptType c : typeDatabase.getClient().query(ConceptType.class)) {
+          conceptTypeRepository.save(modelMapperUtil.mapConceptType(c));
+          count++;
+        }
```
indentation

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/model/ConceptEntry.java:
```diff
@@ -0,0 +1,295 @@
+        return new HashSet<>();
+    }
+    
+    public void setAlternativeIds(String alternativeIds) {
```
relates to my above comment: set implies it replaces whatever is there, not adding to what's already there

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/model/IndexingEvent.java:
```diff
@@ -0,0 +1,102 @@
+
+    public String getLastRun() {
+        if (lastRun == null)
+            return "";
```
code style

jdamerow replied:
?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/IConceptListRepository.java:
```diff
@@ -0,0 +1,24 @@
+ */
+@Repository
+public interface IConceptListRepository extends PagingAndSortingRepository<ConceptList, String> {
+    long deleteByConceptListName(String conceptListName);
```
is this used anywhere?

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/manager/IConceptListManager.java:
```diff
@@ -34,20 +34,19 @@
 	 */
 	public abstract ConceptList getConceptList(String name);
```
is this still being used?

kkrish39 replied:
Yes, Julia. If we try adding concept as a REST request, we are checking if given list name in the JSON body exists/valid.

https://github.com/diging/conceptpower/blob/feature/CCP-248-2/Conceptpower%2BSpring/src/main/java/edu/asu/conceptpower/rest/Concepts.java

jdamerow replied:
That needs to be changed. The only app that uses that endpoint should be Vogon, so it should be ok to change that.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/IConceptListRepository.java:
```diff
@@ -0,0 +1,24 @@
+    
+    List<ConceptList> findAll();
+    
+    ConceptList findByConceptListName(String conceptListName);
```
is this still needed?

kkrish39 replied:
The same reason as above.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java:
```diff
@@ -199,9 +199,9 @@ private void fillConceptEntry(ConceptEntry entry, ConceptsMergeBean conceptMerge
+           entry.getAlternativeIds().addAll(conceptMergeBean.getAlternativeIds());
         } else {
-            entry.setAlternativeIds(conceptMergeBean.getAlternativeIds());
+           entry.setAlternativeIds(String.join(",",conceptMergeBean.getAlternativeIds()));
```
indentation is wrong now I believe?

kkrish39 committed:
[CCP-248] Fixed Indendations

kkrish39 committed:
Fix: Check alternativeIds duplicate values

kkrish39 commented:
The concept list related changes will be done separately as part of CCP-258.

jdamerow commented:
my previous comments beside the concept list issue don't seem to be addressed

kkrish39 commented:
Julia, I've added comments to two those PR reviews. One about the ``` @Transactional ``` annotation in token and user repo. 

For the possible duplication of 'alternativeIds' I've added a check. Please let me know if it's ok.

jdamerow commented:
also, what about the indentation? never mind, just saw that it's outdated.

kkrish39 commented:
Reopening without any changes.

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/IConceptEntryRepository.java:
```diff
@@ -0,0 +1,42 @@
+@Repository
+public interface IConceptEntryRepository extends PagingAndSortingRepository<ConceptEntry, String>, JpaSpecificationExecutor{
+    
+    List<ConceptEntry> findAll();
```
I don't think this method is needed, it should be there by default.

kkrish39 replied:
Yeah, it's there by default. But, the return value is Iterable<>. I've added that as a separate method to return as a List<>.

jdamerow replied:
Well, it's probably returning an iterable, so not all data will be added into memory at once. Why don't you use the iterable?

kkrish39 replied:
Sure we can do that. I changed it to return in List<> because we can avoid changes to the underlying Manager classes consuming this service.

jdamerow replied:
where is that actually needed?

kkrish39 replied:
Umm for instance taking ConceptList, it will be used in Concept wrapper service, Concept Add/Delete controllers which also requires changes to it's dependant class Models.

jdamerow replied:
why do they need to find all concepts? aren't all lists paginated?

kkrish39 replied:
No, the lists aren't paginated

For add/edit concepts, I think we are using it to populate the dropdown

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/IConceptTypeRepository.java:
```diff
@@ -0,0 +1,27 @@
+@Repository
+public interface IConceptTypeRepository extends PagingAndSortingRepository<ConceptType, String>{
+    
+    List<ConceptType> findAll();
```
same as above

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/IIndexRepository.java:
```diff
@@ -0,0 +1,10 @@
+import edu.asu.conceptpower.app.model.IndexingEvent;
+
+public interface IIndexRepository extends PagingAndSortingRepository<IndexingEvent, String>  {
+    List<IndexingEvent> findAll();
```
same as above

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/IReviewRequestRepository.java:
```diff
@@ -0,0 +1,29 @@
+    /*
+     * Return all the review requests from the database
+     * */
+    List<ReviewRequest> findAll();
```
same as above

jdamerow reviewed (changes requested):
I'l continue after the previous comments are resolved.

kkrish39 committed:
[CCP-248] Modified return type for a couple of findAll methods

kkrish39 committed:
[CCP-248] Refactored ConceptEditBean Model

kkrish39 committed:
[CCP-248] Refactored ConceptList and ConceptType repository

kkrish39 committed:
[CCP-248] Minor Refactoring

kkrish39 committed:
[CCP-248] Indendation changes and Minor refactoring

kkrish39 committed:
[CCP-248] Minor refactoring

jdamerow reviewed (changes requested):
- is success.html needed?

jdamerow commented:
Jenkins, run integration tests please

kkrish39 committed:
[CCP-248] Addressing PR comments

kkrish39 commented:
Let open a new PR for this branch as well. The change is not getting reflected here.

![image](https://user-images.githubusercontent.com/55424824/106066718-85ce3780-60ba-11eb-82e5-ff0bafee13dd.png)

I've addressed the alternative's fix and deleted the success.html page.
